### PR TITLE
feat: support date type

### PR DESCRIPTION
**Pull request by fengjiachun**
1. support Date and LocalData
2. version to 0.4.0 (Synchronize with GreptimeDB in major version)

### CODE REVIEW

**Review by MichaelScofield — APPROVED** [review]

File: greptimedb-protocol/src/main/java/io/greptime/models/Util.java [hunk 1]
```diff
@@ -15,17 +15,55 @@
...
+
+        if (value instanceof Number) {
+            return ((Number) value).longValue();
+        }
+
+        // Not null
+        throw new IllegalArgumentException("Unsupported value type: " + value.getClass());
+    }
+
+    static int getDataValue(Object value) {
```
Comment (killme2008) [inline greptimedb-protocol/src/main/java/io/greptime/models/Util.java:46]: typo: `Data` -> `Date`

File: greptimedb-protocol/src/main/java/io/greptime/models/Util.java [hunk 2]
```diff
@@ -15,17 +15,55 @@
...
+        }
+
+        if (value instanceof LocalDate) {
+            return (int) ((LocalDate) value).toEpochDay();
+        }
+
+        return (int) getLongValue(value);
+    }
+
+    static int getDataTimeValue(Object value) {
```
Comment (killme2008) [inline greptimedb-protocol/src/main/java/io/greptime/models/Util.java:60]: typo: `Data` -> `Date`

Comment (fengjiachun) [reply greptimedb-protocol/src/main/java/io/greptime/models/Util.java:60]: Fixed in https://github.com/GreptimeTeam/greptimedb-client-java/pull/58

File: greptimedb-protocol/src/main/java/io/greptime/models/Util.java (hunk 1 above)
Comment (fengjiachun) [reply greptimedb-protocol/src/main/java/io/greptime/models/Util.java:46]: Fixed in https://github.com/GreptimeTeam/greptimedb-client-java/pull/58

**Pull request merged**